### PR TITLE
Update work GitHub Actions

### PR DESCRIPTION
**Pull request by Yozhef**

### CODE REVIEW

**Comment by Yozhef** [comment]
my suggestion to refuse the service flintci / pr - from contributors does not give clarity of check whether they adhered to the project style code, as well as to add a free service codecov to measure test coverage

**Comment by Yozhef** [comment]
@weaverryan any idea how to speed up the review stage?

**Comment by weaverryan** [comment]
Ah, sorry about that! Let me ping my colleague - he's my GH actions expert on our other tutorial - ping @jrushlow :)

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -1,27 +1,43 @@
-name: OAuth2 Client Tests
+name: CI
```
Comment (jrushlow) [inline .github/workflows/ci.yaml:1]: We should keep the name as is for consistency with other Symfony CI Runs

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -1,27 +1,43 @@
...
-    release:
-
-#env:
-#    PHPUNIT_FLAGS: '-v'
-#    SYMFONY_REQUIRE: '>=2.8'
+    push:
+        branches: [ master ]
 
 jobs:
-    composer-validate:
```
Comment (jrushlow) [inline .github/workflows/ci.yaml:15]: The `composer-validate` job needs to stay and should only be executed with the `--strict` flag. I'm ok with this job living in `static-analysis.yaml`

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -1,27 +1,43 @@
...
+                  path: ~/.composer/cache/files
+                  key: ${{ matrix.php }}
+
+            - name: Setup PHP
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: ${{ matrix.php }}
+
+            - name: Validate composer.json
+              run: composer validate --no-check-publish
```
Comment (jrushlow) [inline .github/workflows/ci.yaml:37]: Does not need to run with the matrix

**Review by jrushlow — CHANGES_REQUESTED** [review]
Thanks @Yozhef for implementing the remaining tests. A couple small adjustments and this should be all set!

**Comment by Yozhef** [comment]
@jrushlow thanks corrected

**Review by jrushlow — APPROVED** [review]

**Comment by jrushlow** [comment]
Thanks again @Yozhef!

**Pull request merged**